src/cmipcite/citations.py

```diff
@@ -170,23 +246,34 @@ def get_citations(
 
     Notes
     -----
-     Citation can be retrieved with the help of the Persistent IDentifiers (PIDs).
-     In the CMIP world, there are two types of PIDs:
-       * file PID (also called tracking_id)
-       * dataset PID (often referred to as just PID).
-     A dataset is a collection of files from a single variable sampled at a single
-     frequency from a single model running a single experiment.
-     All datasets from a single model and a single experiment are grouped under a DOI.
-     There exist DOIs associated to single model, but including all the experiments,
-     but they are not used by this package.
+    Citations can be retrieved with the help of the Persistent IDentifiers (PIDs).
+    In the CMIP world, there are two types of PIDs:
+
+       * file PID (normally referred to as a tracking ID)
+       * dataset PID (normally simply referred to as PID).
+
+    A dataset is a collection of files
+    (for CMIP, this collection of files
+    is for a single variable sampled at a single frequency and spatial sampling
+    from a single model running a single experiment).
+    All datasets from a single model and a single experiment
+    are grouped under a DOI, associated with the dataset's PID.
+    There also exist DOIs associated to a single model,
+    that include all the experiments performed by that model,
+    but they are not used by this package at the moment.
     """
-    res = []
-    for input_id in ids_or_paths:
-        # TODO: add checking for and support for paths
-        res.append(
-            get_citation_for_id(
-                input_id, format=format, author_list_style=author_list_style
-            )
+    if client is None:  # pragma: no cover
+        client = RESTHandleClient(handle_server_url="http://hdl.handle.net/")
+
+    doi_versions = [
+        get_doi_and_version(
+            v, client=client, multi_dataset_handling=multi_dataset_handling
         )
+        for v in ids_or_paths
+    ]
+
+    doi_versions_unique = set(doi_versions)
```

Comment:
I would let the user remove the duplicates themselves. If we remove duplicate, it will be harder for the user to associated an input element  with the correct output element. That will be annoying if they want to do anything else after.

Reply:
Would you do that here or introduce a new function? In my head, this function is for, "I have a bunch of IDs, just get me the citations I need for a paper", i.e. de-duplication is desired. The function that supports, "I have a bunch of IDs, get me the mapping between each ID and its citation", is different because a) you're getting a mapping back and b) you get duplicate citations (which isn't useful for e.g. papers).

Reply:
I was imagining a user building a latex table directly from the outputs, but that could be from a different fonction that outputs a dictionnary.